app/src/main/java/com/onegini/mobile/exampleapp/network/fcm/FCMListenerService.java

```diff
@@ -42,6 +56,58 @@ public void onMessageReceived(final RemoteMessage message) {
     }
   }
 
+  @Override
+  public void onNewToken(final String newToken) {
+    super.onNewToken(newToken);
+
+    try {
+      handleTokenUpdate(newToken);
+    } catch (final OneginiInitializationException exception) {
+      setupOneginiSDK(newToken);
```

Comment:
I don't like the fact that we need to check if the SDK was initialised by catching an exception and also the fact that the SDK is already initialised in 3 different places in the app. Maybe we should extract the start methods and error handling to a separate class that returns Completable?

Reply:
Afaik we didn't want to use too much rxjava in the example app, Johan may not use/understand the syntax.

Reply:
I don't see any other way, than catching an exception. We don't have a method `isOneginiSdkSetUp()` atm.